volttron/platform/vip/socket.py

```diff
@@ -85,7 +85,13 @@ def nonblocking(sock):
 
 def encode_key(key):
     '''Base64-encode and return a key in a URL-safe manner.'''
-    assert len(key) in (32, 40)
+    if len(key) % 4 != 0:
+        return key
+    key = key if isinstance(key, bytes) else key.encode("utf-8")
+    try:
+        assert len(key) in (32, 40)
+    except AssertionError:
+        raise AssertionError("Assertion error while encoding key:{}, len:{}".format(key, len(key)))
```

Comment:
Could you explain this logic change? the old behavior was an AssertionError if the length wasn't 32 or 40, but now we return the input with no error if the length isn't a multiple of 4. That seems quite different.

Also, wouldn't it be cleaner to replace lines 91-94 with:
`assert len(key) in (32, 40), f"Assertion error while encoding key:{key}, len:{len(key)}"`

Reply:
done